pkg/types/chain_reader.go

```diff
@@ -70,6 +74,7 @@ type Sequence struct {
 type BoundContract struct {
```

Comment:
I think a method on BoundContract that returns a single identifier is very useful since codec contract types are identified by contractName + methodName

Reply:
Is that not what `Name` is? The single identifier?

Reply:
Ah right, the types would be the same for every address, so name is fine